modules/javafx.graphics/src/main/java/com/sun/glass/ui/WindowControlsOverlay.java

```diff
@@ -0,0 +1,539 @@
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.glass.ui;
+
+import com.sun.glass.events.MouseEvent;
+import com.sun.javafx.util.Utils;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.ReadOnlyObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.value.ObservableValue;
+import javafx.css.CssMetaData;
+import javafx.css.PseudoClass;
+import javafx.css.SimpleStyleableBooleanProperty;
+import javafx.css.SimpleStyleableIntegerProperty;
+import javafx.css.SimpleStyleableObjectProperty;
+import javafx.css.StyleConverter;
+import javafx.css.Styleable;
+import javafx.css.StyleableBooleanProperty;
+import javafx.css.StyleableIntegerProperty;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableProperty;
+import javafx.geometry.Dimension2D;
+import javafx.geometry.HPos;
+import javafx.geometry.HorizontalDirection;
+import javafx.geometry.NodeOrientation;
+import javafx.geometry.VPos;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.layout.Region;
+import javafx.scene.paint.Paint;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.util.Subscription;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Contains the visuals and behaviors for the minimize/maximize/close buttons on an {@link StageStyle#EXTENDED}
+ * window for platforms that use client-side decorations (Windows and Linux/GTK). This control supports
+ * left-to-right and right-to-left orientations, as well as a customizable layout order of buttons.
+ *
+ * <h2>Substructure</h2>
+ * <ul>
+ *     <li>{@link Region} — {@code window-button-container}
+ *     <ul>
+ *         <li>{@link Region} — {@code window-button}, {@code minimize-button}
+ *         <li>{@link Region} — {@code window-button}, {@code maximize-button}
+ *         <li>{@link Region} — {@code window-button}, {@code close-button}
+ *     </ul>
+ * </ul>
+ *
+ * <table style="white-space: nowrap">
+ *     <caption>CSS properties of {@code window-button-container}</caption>
```

Comment:
It can't be customized by users, as there is no API for that. There wouldn't be a platform-independent way to do that, because we're not using `WindowControlsOverlay` on macOS in the first place.